### PR TITLE
Add functions with `dims` keyword argument

### PR DESCRIPTION
**Pull request by martinholters**
Add (unexported) `Compat.accumulate`, `Compat.accumulate!`,
`Compat.all`, `Compat.any`, `Compat.cor`, `Compat.cov`,
`Compat.cumprod`, `Compat.cumprod!`, `Compat.cumsum`, `Compat.cumsum!`,
`Compat.findmax`, `Compat.findmin`, `Compat.mapreduce`,
`Compat.maximum`, `Compat.mean`, `Compat.median`, `Compat.minimum`,
`Compat.prod`, `Compat.reduce`, `Compat.sort`, `Compat.std`,
`Compat.sum`, `Compat.var`, and `Compat.varm` with `dims` keyword
argument.

### CODE REVIEW

**Comment by martinholters** [comment]
If we wait for https://github.com/JuliaLang/julia/pull/26521, I can put the commented tests behind `VERSION` conditionals, exempting only the version range where Julia was broken. (Of course, we can also merge this now and fix up the tests later.)

**Comment by fredrikekre** [comment]
I added the version check. I didn't verify this, but you implied in the julia issue that it was broken by https://github.com/JuliaLang/julia/pull/25989?

File: src/Compat.jl [hunk 1]
```diff
@@ -1690,6 +1690,72 @@ if VERSION < v"0.7.0-DEV.4585"
...
+        dims===nothing ? Base.cor(a, b) : Base.cor(a, b, dims)
+    @eval mapreduce(f, op, a::AbstractArray; dims=nothing) =
+        dims===nothing ? Base.mapreduce(f, op, a) : Base.mapreducedim(f, op, a, dims)
+    @eval mapreduce(f, op, v0, a::AbstractArray; dims=nothing) =
+        dims===nothing ? Base.mapreduce(f, op, v0, a) : Base.mapreducedim(f, op, a, dims, v0)
+    @eval reduce(op, a::AbstractArray; dims=nothing) =
+        dims===nothing ? Base.reduce(op, a) : Base.reducedim(op, a, dims)
+    @eval reduce(op, v0, a::AbstractArray; dims=nothing) =
+        dims===nothing ? Base.reduce(op, v0, a) : Base.reducedim(op, a, dims, v0)
+    @eval accumulate!(op, out, a; dims=nothing) =
```
Comment (fredrikekre) [inline src/Compat.jl]: I guess these `@eval` could be removed? Only the ones in the loops over function symbols is needed AFAICT.

**Comment by martinholters** [comment]
> I added the version check.

Thanks, but I've actually bisected the origin to an earlier change; the correct lower bound would have to be v"0.7.0-DEV.3262". I'll update accordingly.

File: src/Compat.jl (hunk 1 above)
Comment (martinholters) [reply src/Compat.jl]: Good catch. I'll update.

Comment (martinholters) [reply src/Compat.jl]: Um, no, that gives
```
ERROR: LoadError: error compiling anonymous: error in method definition: function Base.accumulate! must be explicitly imported to be extended
```
There is no other use `accumulate!` though, so why doesn't it just create a new binding to a new function? BUT if I move the definition to its own `if` block, it does work without the `@eval`. Strange...

Comment (fredrikekre) [reply src/Compat.jl]: Yea I tried the same, lets just leave it as is I guess.

**Pull request merged**